spring-integration-webflux/src/main/java/org/springframework/integration/webflux/outbound/WebFluxRequestExecutingMessageHandler.java

```diff
@@ -169,13 +188,35 @@ protected Object exchange(Supplier<URI> uriSupplier, HttpMethod httpMethod, Http
 										ResponseEntity.status(response.statusCode())
 												.headers(response.headers().asHttpHeaders());
 
-								Mono<?> bodyMono = Mono.empty();
-
-								if (expectedResponseType instanceof ParameterizedTypeReference<?>) {
-									bodyMono = response.body(BodyExtractors.toMono((ParameterizedTypeReference<?>) expectedResponseType));
+								Mono<?> bodyMono;
+
+								if (expectedResponseType != null) {
```

Comment:
Is a check made somewhere to see if the response is 2xx? Should we be trying to parse the response if it's not?

Reply:
Yes, we have that one there:
```
if (httpStatus.isError()) {
	return response.body(BodyExtractors.toDataBuffers())
			.reduce(DataBuffer::write)
                         ...
			.map(bodyBytes -> {
					throw new WebClientResponseException(...);
					}
			);
}
else {
	return Mono.just(response);
}
```
See `WebFluxRequestExecutingMessageHandler.exchange()` source code.